plugins/GSdx/GSOsdManager.cpp

```diff
@@ -208,104 +234,119 @@ void GSOsdManager::RenderString(GSVertexPT1* dst, std::u32string msg, float x, f
-		dst += pair.second.size() * 6;
-		drawn += pair.second.size() * 6;
-	}
+		for(auto it = m_log.begin(); it != m_log.end();) {
+			if(it->msg.size() * 6 > count - drawn) break;
 
-	float offset = 0;
+			float x = -1 + 8 * m_sx;
+			float y = 1 - ((m_size+2)*(it-m_log.begin()+1)) * m_sy;
+			if(y + offset < -1) break;
 
-	for(auto it = m_log.begin(); it != m_log.end();) {
-		if(it->msg.size() * 6 > count - drawn) break;
+			if(it->OnScreen.time_since_epoch().count() == 0)
+				it->OnScreen = std::chrono::system_clock::now();
 
-		float x = -1 + 8 * m_sx;
-		float y = 1 - ((m_size+2)*(it-m_log.begin()+1)) * m_sy;
-		if(y + offset < -1) break;
+			std::chrono::duration<float> elapsed = std::chrono::system_clock::now() - it->OnScreen;
+			if(elapsed > std::chrono::seconds(6)) {
+				it = m_log.erase(it);
+				continue;
+			}
 
-		if(it->OnScreen.time_since_epoch().count() == 0)
-			it->OnScreen = std::chrono::system_clock::now();
-
-		std::chrono::duration<float> elapsed = std::chrono::system_clock::now() - it->OnScreen;
-		if(elapsed > std::chrono::seconds(6)) {
-			it = m_log.erase(it);
-			continue;
-		}
-
-		uint8 alpha = 0xFF;
-		if(elapsed > std::chrono::seconds(3)) {
 			float ratio = (elapsed - std::chrono::seconds(3)) / std::chrono::seconds(3);
-			ratio = ratio > 1 ? 1 : ratio;
-			alpha = 0xFF * (1 - ratio);
-			offset += ((m_size+2) * m_sy) * ratio;
+			ratio = ratio > 1.0 ? 1.0 : ratio < 0.0 ? 0.0 : ratio;
+
+			y += offset += ((m_size+2) * m_sy) * ratio;
+			uint32 color = it->color;
+			((uint8 *)&color)[3] = ((uint8 *)&color)[3] * (1.0 - ratio) * transparency;
+			RenderString(dst, it->msg, x, y, m_sx, m_sy, color);
+			dst += it->msg.size() * 6;
+			drawn += it->msg.size() * 6;
+			++it;
 		}
+	}
 
-		y += offset;
-		uint32 color = it->color;
-		((uint8 *)&color)[3] = (((uint8 *)&color)[3] * alpha) / 0xFF;
-		RenderString(dst, it->msg, x, y, m_sx, m_sy, color);
-		dst += it->msg.size() * 6;
-		drawn += it->msg.size() * 6;
-		++it;
+	if(theApp.GetConfigB("monitor_enabled")) {
+		size_t line = 0;
+
+		for(const auto &pair : m_monitor) {
+			if((pair.first.size() + pair.second.size()) * 6 > count - drawn) break;
+
+			float x = 1.0 - (8 + m_max_width * (pair.first.size() + 1 + pair.second.size())) * m_sx;
+			float y = -0.99 + ((m_size+2)*m_sy) * line++;
+			uint32 color = 0xFFFF00FF;
+			((uint8 *)&color)[3] = ((uint8 *)&color)[3] * transparency;
+			RenderString(dst, pair.first, x, y, m_sx, m_sy, color);
+			dst += pair.first.size() * 6;
+			drawn += pair.first.size() * 6;
+
+			x = 1.0 - (8 + m_max_width * pair.second.size()) * m_sx;
+			color = 0xFFFF00FF;
+			((uint8 *)&color)[3] = ((uint8 *)&color)[3] * transparency;
+			RenderString(dst, pair.second, x, y, m_sx, m_sy, color);
+			dst += pair.second.size() * 6;
+			drawn += pair.second.size() * 6;
+		}
 	}
 
-	for(const auto pair : m_indicator) {
-		uint8 alpha = 0xFF;
-		if(!pair.second.on) {
-			alpha /= 4;
+	if(theApp.GetConfigB("indicator_enabled")) {
```

Comment:
FYI, GetConfig call are quite slow, it is often to cache them in a local/member  variables